packages/hash/api/src/graphql/typeDefs/paginationConnections.typedef.ts

```diff
@@ -0,0 +1,66 @@
+import { gql } from "apollo-server-express";
+
+export const pagePaginationTypedef = gql`
+  """
+  Each pagination specifies if there are more results to request.
+  Please use the 'nextPageCursor' for every subsequent pagination request with the 'after' parameter.
+  """
+  type PageInfo {
+    # could also just be a nextPageCursor since its presence encodes the boolean
+    # kept in to comply with spec.
+    hasNextPage: Boolean!
+    """
+    Number of pages that can be paginated.
+    This number reflects total result count divided by page size.
+    """
+    pageCount: Int!
+    """
+    To be used for every subsequent pagination request.
+    """
+    nextPageCursor: String
+  }
+
+  """
+  There are no cursors per-edge currently.
+  """
+  type PageSearchResultEdge {
```

Comment:
We might want to make this a more generic `SearchResult` edge when introducing more OpenSearch-using queries - but let's worry about that then.